scripts/copyfiles.py

```diff
@@ -66,17 +66,41 @@ def main(log_level, source_dir: str, target_dir: str, filenames: list[str]):
             datefmt='%Y-%m-%d %H:%M:%S'
         )
 
+    # code ALWAYS uses absolute paths since we replace path prefixes
+    source_dir = os.path.abspath(source_dir)
+    target_dir = os.path.abspath(target_dir)
+
     # We are copying to target ...
     if not os.path.exists(target_dir):
         logging.info("Creating output directory %s", target_dir)
         os.makedirs(target_dir)
 
     for filename in filenames:
+        # Take the relative path from the filename into the source
+        #
+        # Example if:
+        #   source_dir  => "/SOME/ABSOLUTE/PATH"
+        #   filename  => "/SOME/ABSOLUTE/PATH/that/is/used.h"
+        #
+        # then:
+        #  relative_path => "that/is/used.h"
+        #
+        filename = os.path.abspath(filename)
         relative_path = os.path.relpath(filename, source_dir)
         if not filename.endswith(relative_path):
             logging.error("%s does not seem relative to %s: relative path is %s", filename, source_dir, relative_path)
             sys.exit(1)
 
+        # Prepend the destination directory to relative path. So that if we have:
+        #
+        # Example if:
```

Comment:
What happens if you have:
```
        #   source_dir  => "/SOME/ABSOLUTE/PATH"
        #   target_dir  => "/OTHER/LOCATION"
        #   filename  => "/SOME/ABSOLUTE/DIFFERENT_PATH/that/is/used.h"
```

Won't relative_path be `../DIFFERENT_PATH/that/is/used.h`

If the expectation is that filename won't be outside of `source_dir` then I am okay with simply checking that `..` doesn't exist in `relative_path` and erroring out

Reply:
The expectation is that filename must be inside source dir. I added some partial check in line 90 as `if not filename.endswith(relative_path):` which seems to work:

![image](https://github.com/user-attachments/assets/d948352f-016f-4132-a919-428fb1c00e3c)

Reply:
Actually running like that:

![image](https://github.com/user-attachments/assets/fa765604-b2c3-45a8-b4f6-800d3856f38d)

Reply:
I adjusted the message a bit to say "not a child of" and enabled error output by default, so these errors are more obvious (not just the error return code).